test/helpers/kubectl.go

```diff
@@ -3833,14 +3837,19 @@ func (kub *Kubectl) HelmAddCiliumRepo() *CmdRes {
 }
 
 // HelmTemplate renders given helm template. TODO: use go helm library for that
+// We use --validate with `helm template` to properly populate the built-in objects like
+// .Capabilities.KubeVersion with the values from associated cluster.
+// This comes with a caveat that the command might fail if helm is not able to validate the
+// chart install on the cluster, like if a previous cilium install is not cleaned up properly
+// from the cluster.
```

Comment:
What happens if it fails?

Reply:
The caller has to make sure that there are no leftover cilium components in the cluster. I will add this line to the documentation of the function.